### PR TITLE
[3732] Add new tool for encrypting text

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/3732

### CODE REVIEW

File: static-assets/components/cstudio-admin/base.js [hunk 1]
```diff
@@ -209,12 +209,18 @@ CStudioAdminConsole.Toolbar.prototype = {
     YAHOO.util.Event.on(toolContainerEl, 'click', onRenderWorkAreaFn, {tool: tool, toolbar: this});
 
     this.tools[this.tools.length] = tool;
+
+    // TODO: To be removed once development is over
```
Comment (rart) [inline static-assets/components/cstudio-admin/base.js]: 🤦🏻‍♂️

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-admin/base.js (hunk 1 above)
Comment (jvega190) [reply static-assets/components/cstudio-admin/base.js]: done

File: static-assets/components/cstudio-admin/mods/admin-configurations.js [hunk 2]
```diff
@@ -15,476 +15,606 @@
...
+        var xml = editor.getValue();
+        var savePath = itemSelectEl[selectedIndex].value;
+        if (savePath != 'undefined' && savePath != '') {
+
+          var defPath = itemSelectEl[selectedIndex].value;
+
+          var url = "/api/2/configuration/write_configuration";
+
+          try {
+            parseValidateDocument(xml);
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/admin-configurations.js]: Don’t see where xml is declared inside this function ?

File: static-assets/components/cstudio-admin/mods/admin-configurations.js [hunk 3]
```diff
@@ -15,476 +15,606 @@
...
+        try {
+          const doc = parseValidateDocument(value).documentElement;
+          const tags = doc.querySelectorAll('[encrypted]');
+          const items = findPendingEncryption(tags);
+          if (items.length) {
+            editor.setOption('readOnly', true);
+            editor.container.style.opacity = .5;
+            // Not sure this is supported in all relatively relevant
+            // browsers, test and change or simply change this syntax to es5
+            const { rxjs: { forkJoin, operators: { map } }, util: { ajax } } = CrafterCMSNext;
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/admin-configurations.js:493]: Did you test this syntax on firefox and safari at minimum? but preferably edge too.

File: static-assets/components/cstudio-admin/mods/admin-configurations.js [hunk 4]
```diff
@@ -15,476 +15,606 @@
...
+            forkJoin(
+              items.map(({ tag, text }) =>
+                ajax.get(`/studio/api/2/security/encrypt.json?text=${text}`).pipe(
+                  map(({ response }) => ({ tag, text: response.item }))
+                )
+              )
+            ).subscribe(
+              (encrypted) => {
+                encrypted.forEach(({ text, tag }) => {
+                  tag.innerHTML = text;
```
Comment (rart) [inline static-assets/components/cstudio-admin/mods/admin-configurations.js]: Per @jdrossl, the value needs to be wrapped in `${enc:VALUE}`.

Change to 
```js
tag.innerHTML = `\${enc:${text}}`;
```

**Review by rart — APPROVED** [review]

**Review by avasquez614 — APPROVED** [review]

**Pull request merged**